runelite-client/src/main/java/net/runelite/client/plugins/playerindicators/PlayerIndicatorsOverlay.java

```diff
@@ -80,7 +80,7 @@ private void renderPlayerOverlay(Graphics2D graphics, Player actor, Color color)
 		int offset = actor.getLogicalHeight() + 40;
 		Point textLocation = actor.getCanvasTextLocation(graphics, name, offset);
 
-		if (textLocation != null)
+		if (textLocation != null && config.drawNames())
```

Comment:
You should probably move this boolean check before `String name = actor.getName().replace('\u00A0', ' ');` as early return